serde_arrow/benches/groups/json_to_arrow.rs

```diff
@@ -0,0 +1,101 @@
+use crate::groups::complex_common::Item;
+
+use {
+    serde_arrow::schema::{SchemaLike as _, SerdeArrowSchema},
+    std::sync::Arc,
+};
+
+// arrow-version:replace: use arrow_json_{version}::ReaderBuilder;
+use arrow_json_50::ReaderBuilder;
+
+// arrow-version:replace: use arrow_schema_{version}::Schema as ArrowSchema;
+use arrow_schema_50::Schema as ArrowSchema;
+
+// arrow-version:replace: use arrow_array_{version}::RecordBatch;
+use arrow_array_50::RecordBatch;
+
+fn benchmark_json_to_arrow(c: &mut criterion::Criterion) {
+    let rng = &mut rand::thread_rng();
+    let items = (0..10_000)
+        .map(|_| Item::random(rng))
+        .collect::<Vec<Item>>();
+    let jsons_to_deserialize = items
+        .iter()
+        .map(|item| serde_json::to_string(item).expect("Failed to serialize JSON"))
+        .collect::<Vec<_>>();
+    let jsons_to_deserialize_concatenated = jsons_to_deserialize.join("\n");
+    let jsons_to_deserialize: Vec<&str> = {
+        let mut prev = 0;
+        jsons_to_deserialize
+            .iter()
+            .map(|s| {
+                let ret = &jsons_to_deserialize_concatenated[prev..(prev + s.len())];
+                prev += s.len() + 1;
+                ret
+            })
+            .collect::<Vec<_>>()
+    };
+    let schema = SerdeArrowSchema::from_samples(&items, Default::default()).unwrap();
+    let arrow_fields = schema.to_arrow_fields().unwrap();
+    let mut group = c.benchmark_group(format!("json_to_arrow({})", items.len()));
+
+    // arrow-json direct
+    group.bench_function("arrow-json", |b| {
+        b.iter(|| {
+            let schema = Arc::new(ArrowSchema::new(arrow_fields.to_owned()));
+            let mut decoder = ReaderBuilder::new(schema.clone()).build_decoder().unwrap();
+            decoder
+                .decode(criterion::black_box(
+                    jsons_to_deserialize_concatenated.as_bytes(),
+                ))
+                .unwrap();
+            let arrays = decoder.flush().unwrap().unwrap().columns().to_vec();
+            let record_batch = RecordBatch::try_new(schema, arrays).unwrap();
+            record_batch
+        })
+    });
+
+    // arrow-json via serde
+    group.bench_function("serde_json_transcode_arrow-json", |b| {
+        b.iter(|| {
+            let schema = Arc::new(ArrowSchema::new(arrow_fields.to_owned()));
```

Comment:
I also coded up a quick benchmark in between, but was using serde_json::Value instead of transcoders. My impression was that the performance was slightly better than the transcode impl. 

For reference that was my benchmark impl:

<details>

```rust
pub fn benchmark_fn(c: &mut criterion::Criterion) {
    let data = (0..10_000).map(|_| Item::random(&mut rand::thread_rng())).collect::<Vec<_>>();

    let mut json_lines = Vec::<u8>::new(); 
    data.iter().for_each(|item| {
        serde_json::to_writer(&mut json_lines, item).unwrap();
        writeln!(&mut json_lines).unwrap();
    });
    
    let schema = SerdeArrowSchema::from_type::<Item>(Default::default()).unwrap();
    let arrow_fields = schema.to_arrow_fields().unwrap();

    let mut group = c.benchmark_group("json_to_arrow");
    group.sample_size(20);
    group.sampling_mode(criterion::SamplingMode::Flat);
    group.measurement_time(std::time::Duration::from_secs(120));

    group.bench_function("serde_arrow", |b| {
        b.iter(|| {
            let mut items = Vec::new();
            data_json.split(|c| *c == b'\n').for_each(|item| {
                if item.len() != 0 {
                    let item: Value = serde_json::from_slice(item).unwrap();
                    items.push(item);
                }
            });
            let res = serde_arrow::to_arrow(&arrow_fields, &items).unwrap();
            criterion::black_box(res);
        });
    });

    group.bench_function("arrow_json", |b| {
        b.iter(|| {
            use serde_arrow::_impl::arrow::_raw::schema::{Schema, ArrowError};
            use arrow_json_50::ReaderBuilder;

            let data = Cursor::new(&json_lines);
            let reader = ReaderBuilder::new(Arc::new(Schema::new(arrow_fields.clone()))).build(data).unwrap();
            let res = reader.collect::<Result<Vec<_>, ArrowError>>().unwrap();
            criterion::black_box(res);
        });
    });
}
```

</details>